### DISCUSSION TITLE
persistent SELECT 1 error

### DISCUSSION THREAD

**Discussion by espinielli** [trinodb/trino-python-client#456] [Q&A]
Every time I execute a query on trino (via a python binding) I get the error below

```
EXECUTE IMMEDIATE not available for trino.opensky-network.org:443; defaulting to legacy prepared statements (TrinoUserError(type=USER_ERROR, name=SYNTAX_ERROR, message="line 1:19: mismatched input ''SELECT 1''. Expecting: 'USING', <EOF>", query_id=20240318_210214_00552_duudt))

```

I traced it down to line 286 in https://github.com/trinodb/trino-python-client/blob/master/trino/dbapi.py#L272-L290
but Ihave no idea why it is executed...
Anybody is experiencing this or has any hints at what is going on?

the trino version is
```
trino> select version();
 _col0
-------
 408
(1 row)
```

**Comment by hovaesco** [comment]
You're running an older Trino version which doesn't support `EXECUTE IMMEDIATE`. You can either upgrade your Trino version or pass `legacy_prepared_statements=True` along other connection parameters.

**Reply by espinielli** [reply]
Thanks @hovaesco 
just to be sure I understood correctly, you mean to use something like (partially copied from [trino-python-client README](https://github.com/trinodb/trino-python-client/blob/master/README.md))

```python
from sqlalchemy import create_engine
from trino.sqlalchemy import URL

engine = create_engine(
    URL(
        host="localhost",
        port=8080,
        catalog="system"
    ),
    connect_args={
      "legacy_prepared_statements": "True"
    }
)
```
?

**Comment by hovaesco** [comment]
If you're using SQLAlchemy then yes, if not then just follow DBAPI docs https://github.com/trinodb/trino-python-client?tab=readme-ov-file#the-python-database-api-dbapi
